server/server.go

```diff
@@ -311,13 +314,13 @@ func New(cfg *config.Config) (*Server, error) {
 	s.log.Noticef("Server link public key hash is: %x", linkPubKeyHash[:])
 
 	if s.cfg.Debug.GenerateOnly {
-		return nil, ErrGenerateOnly
+		return ErrGenerateOnly
```

Comment:
No. We have to actually abide by the function signature and return an error. Setting err is meaningless otherwise. I think what you are remembering here is that sometimes go functions have a signature that specifies the return type with variable names. This is not one of those cases.